src/mist/internal/http.gleam

```diff
@@ -224,15 +224,15 @@ pub fn parse_request(
         |> bit_string.to_string
         |> result.replace_error(InvalidPath),
       )
-      let #(path, query) = case string.split(path, "?") {
-        [path] -> #(path, [])
-        [path, query_string] -> {
+      let #(path, query) = case string.split_once(path, "?") {
```

Comment:
This is okay, but I wonder if this should just be using `uri.parse`?  That does work with just `/a/b?query=true`.  We can pluck out `Uri(path: path, query: query, ..)` and use `uri.parse_query` as below.

Reply:
I guess that could be done. I am not sure if that affects how `#` is handled in the request line and if they have the same meaning in request lines as they do in URLs.

I'm not sure why `parse_query` is necessary here, since `Request.query` is just an `Option(String)`. This parsing round trip does not produce the same value again if you keep in mind that a URL query is not necessarily `x-www-form-urlencoded`.

For example, the following request will have a mangled query parameter: `/a?query,test` will be turned into `/a?query,test=` after passing through this code. If I wanted to use a CSV line in my query for some reason, that would change the content. This is due to the way `request.set_query` works, which assumes the query is `x-www-form-urlencoded`. This can be bypassed by passing the query to the `Request` constructor directly.

Reply:
Ah gotcha, that makes sense. Thanks for clarifying! I'm also fine to remove the `parse_query` call below. That does seem like something that the consumer for handle.

Reply:
Should be fixed now. Also added another test for what I described above.